ui/src/features/freight-timeline/freight-timeline-header.tsx

```diff
@@ -65,6 +66,28 @@ export const FreightTimelineHeader = ({
 
   const navigate = useNavigate();
 
+  const WarehouseSelector = (
+    <Select
+      className='w-48 ml-4'
+      value={selectedWarehouse}
+      onChange={(value) => setSelectedWarehouse(value)}
+      size='small'
+      labelRender={({ label }) => <div className='text-xs font-medium'>{label}</div>}
+      optionRender={(opt) => (
+        <div className='text-xs font-normal w-full h-full flex items-center'>
+          <div>{opt.label}</div>
+        </div>
+      )}
+      options={[
+        ...(Object.keys(warehouses) || []).map((w) => ({ value: w, label: w })),
```

Comment:
if the warehouses variable is undefined, this OR operator doesn't make sense here, I guess. It just throws an error.

Reply:
the default is not undefined or null but empty object thats for sure